### PR TITLE
Replace regex parser - use re.search instead re.match

### PR DESCRIPTION
**Pull request by FabioRosado**
# Description
As @gabeduke mentioned in gitter, the regex parser is ignoring the regular expression to be searched if it's not at the beginning of the string. After a few testing and reading it seems that the reason why this is happening because re.match will only match a regular expression if ifound at the beginning of the string.

Using re.search seems to fix this issue as this method will search through the whole string for the regular expression specified on the skill.  So I've replaced `re.match` with `re.search` in opsdroid.parsers.regex

I've also updated the documentation to show the importance of using anchors when building skills, please let me know if the examples make sense of if they are too confusing.

I've also replaced line 94 of opsdroid.core: `print("Error: " + error)`. When spaces are not used properly in configuration.yaml the following exception is returned:

```python
  File "/Users/fabiorosado/Documents/GitHub/opsdroid/opsdroid/core.py", line 94, in critical
    print("Error:" + error)
TypeError: must be str, not ParserError
```

By replacing concatenation with a comma the print statement works normally. I've noticed that we are currently logging this, do we still need this print?

**Fixes:** N/A

## Status
**READY** | ~~**UNDER DEVELOPMENT**~~ | ~~**ON HOLD**~~

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [X] Documentation (fix or adds documentation)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Tested both locally and Travis CI - All tests passed
- [X] Tested different skills to see if any side-effects occur - none experienced

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/matchers/regex.md [hunk 1]
```diff
@@ -14,6 +16,52 @@ async def hello(opsdroid, config, message):
...
+#### Fixed example
+```python
+from opsdroid.matchers import match_regex
+
+@match_regex('cold$')
+async def is_cold(opsdroid, config, message):
+    await message.respond('it is')
+```
+
+Now this skill will only be triggered if `'cold'` is located at the end of the message.
```
Comment (jacobtomlinson) [inline docs/matchers/regex.md]: This wouldn't actually work as the end of the message is `cold?` not `cold`. You either need to remove the question mark from the example or add the question mark to the regex. I'd probably remove the question mark.

File: opsdroid/core.py [hunk 2]
```diff
@@ -92,7 +92,7 @@ def critical(self, error, code):
         """Exit due to unrecoverable error."""
         self.sys_status = code
         _LOGGER.critical(error)
-        print("Error: " + error)
+        print("Error:", error)
```
Comment (jacobtomlinson) [inline opsdroid/core.py]: Thanks for fixing this! I know it's only a tiny change but this probably should've been in a separate PR. I'll leave it for now, but something to bear in mind in the future.

**Comment by FabioRosado** [comment]
Thanks for that Jacob I didn't think about the question mark when I wrote the example. I've removed it from the example so things should look okay now

**Comment by jacobtomlinson** [comment]
Nice one!

**Pull request merged**